File: src/test/java/boat/torrent/ZooqleTest.java
package boat.torrent;

import java.util.List;

import boat.utilities.HttpHelper;
import org.junit.jupiter.api.Test;

import static org.junit.jupiter.api.Assertions.assertDoesNotThrow;
import static org.junit.jupiter.api.Assertions.assertEquals;

class ZooqleTest {

    @Test
    void shouldFindTorrents() {
        // Given
        // When
        assertDoesNotThrow(() -> new Zooqle(new HttpHelper()).searchTorrents("planet"));
    }

    @Test
    void shouldExtractSize() {
        // Given
        // When
        assertDoesNotThrow(() -> new Zooqle(new HttpHelper()).searchTorrents("blackish+s02e09"));
    }

    @Test
    void shouldFindNoTorrents() {
        // Given
        // When
        List<Torrent> torrentList = new Zooqle(new HttpHelper()).searchTorrents("ThisshouldntbeafindableStringAtall");
        // Then
        assertEquals(0, torrentList.size());
    }

}

File: src/test/java/boat/torrent/ZooqleTest.kt
package boat.torrent

import boat.utilities.HttpHelper
import org.junit.jupiter.api.Assertions
import org.junit.jupiter.api.Disabled
import org.junit.jupiter.api.Test

internal class ZooqleTest {
    @Disabled
    @Test
    fun shouldFindTorrents() {
        // Given
        // When
        Assertions.assertDoesNotThrow<List<Torrent>> { Zooqle(HttpHelper()).searchTorrents("planet") }
    }

    @Disabled
    @Test
    fun shouldFindNoTorrents() {
        // Given
        // When
        val torrentList = Zooqle(HttpHelper()).searchTorrents("ThisshouldntbeafindableStringAtall")
        // Then
        Assertions.assertEquals(0, torrentList.size)
    }
}